vcn_gateways.tf

```diff
@@ -23,8 +23,52 @@ resource "oci_core_route_table" "ig" {
   freeform_tags = var.tags
 
   route_rules {
+    # * With this route table, Internet Gateway is always declared as the default gateway
     destination       = local.anywhere
     network_entity_id = oci_core_internet_gateway.ig[0].id
+    description       = "Terraformed - Auto-generated at Internet Gateway creation: Internet Gateway as default gateway"
+  }
+
+  dynamic "route_rules" {
+    # * filter var.internet_gateway_route_rules for routes with "drg" as destination
+    # * and steer traffic to the module created DRG
+    for_each = var.internet_gateway_route_rules != null ? { for k, v in var.internet_gateway_route_rules : k => v
```

Comment:
you also need to add a condition to add this rule if the DRG is created

Reply:
This block will be evaluated only if the parent resource (route table) is created, which itself have a test to be created based on `var.service_gateway_enabled` value.

I don't think we have to check `var.service_gateway_enabled` value again for dynamic block creation.

Reply:
This is on internet_gateway, not service_gateway. What happens when you create the IGW but not the DRG?

Reply:
inside the same route table, each use case is evaluated by a different dynamic block. No dependency regarding which gateways you choose to deploy.

You can create one , two or all, the rules will reflect what you chose.